api/src/scripts/populate_db_gtfs.py

```diff
@@ -236,40 +244,41 @@ def trigger_downstream_tasks(self):
         env = os.getenv("ENV")
         self.logger.info(f"ENV = {env}")
         if os.getenv("ENV", "local") != "local":
-            publish_all(self.added_gtfs_feeds)  # Publishes the new feeds to the Pub/Sub topic to download the datasets
+            # Publishes the new feeds to the Pub/Sub topic to download the datasets
+            publish_all(self.added_gtfs_feeds)
 
-    def post_process_locations(self):
+    def post_process_locations(self, session: "Session"):
         """
         Set the country for any location entry that does not have one.
         """
-        query = self.db.session.query(Location).filter(Location.country.is_(None))
+        query = session.query(Location).filter(Location.country.is_(None))
         result = query.all()
         set_country_count = 0
         for location in result:
             country = self.get_country(location.country_code)
             if country:
                 location.country = country  # Set the country field to the desired value
                 set_country_count += 1
-        self.db.session.commit()
+        session.commit()
         self.logger.info(f"Had to set the country for {set_country_count} locations")
 
     # Extracted the following code from main, so it can be executed as a library function
     def initialize(self, trigger_downstream_tasks: bool = True):
         try:
             configure_polymorphic_mappers()
-            self.populate_db()
-            self.db.session.commit()
+            with self.db.start_db_session() as session:
+                self.populate_db(session)
+                session.commit()
 
-            self.logger.info("Refreshing MATERIALIZED FEED SEARCH VIEW - Started")
-            self.db.session.execute(text(f"REFRESH MATERIALIZED VIEW CONCURRENTLY {t_feedsearch.name}"))
-            self.logger.info("Refreshing MATERIALIZED FEED SEARCH VIEW - Completed")
-            self.db.session.commit()
-            self.logger.info("\n----- Database populated with sources.csv data. -----")
-            if trigger_downstream_tasks:
-                self.trigger_downstream_tasks()
+                self.logger.info("Refreshing MATERIALIZED FEED SEARCH VIEW - Started")
+                session.execute(text(f"REFRESH MATERIALIZED VIEW CONCURRENTLY {t_feedsearch.name}"))
+                self.logger.info("Refreshing MATERIALIZED FEED SEARCH VIEW - Completed")
+                session.commit()
+                self.logger.info("\n----- Database populated with sources.csv data. -----")
+                if trigger_downstream_tasks:
+                    self.trigger_downstream_tasks()
         except Exception as e:
             self.logger.error(f"\n------ Failed to populate the database with sources.csv: {e} -----\n")
-            self.db.session.rollback()
```

Comment:
[question]: should we rollback here?

Reply:
rollback will be called inside contextmanager [here](https://github.com/MobilityData/mobility-feed-api/blob/bae94e06f77ddc49bd8672b6275097d2e45ec4e1/api/src/database/database.py#L121-L122)
```Python
    @contextmanager
    def start_db_session(self):
        session = self.Session()
        try:
            yield session
            session.commit()
        except Exception:
            session.rollback()
            raise
        finally:
            session.close()
```

Reply:
I got it; thanks for the clarification.